src/fetchers/tarball-fetcher.js

```diff
@@ -208,11 +216,7 @@ export default class TarballFetcher extends BaseFetcher {
       return this.fetchFromLocal(this.reference);
     }
 
-    if (await this.getLocalAvailabilityStatus()) {
-      return this.fetchFromLocal();
-    } else {
-      return this.fetchFromExternal();
-    }
+    return this.fetchFromLocal().catch(err => this.fetchFromExternal());
```

Comment:
Don't agree with this part is what the old code did mostly and `fetchFromLocal` doesn't actually throw `ENOENT` errors. It throws handmade "can't find file in cache" errors. That said we can check specifically for `ENOENT` errors inside `fetchFromLocal` trials but I'm not really sure if it matters since if it is something else, we still ended up not being able to read the file and we want to try the next file name.